packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/class_filter_dialog.dart

```diff
@@ -0,0 +1,159 @@
+// Copyright 2022 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import '../../../../../shared/common_widgets.dart';
+import '../../../../../shared/dialogs.dart';
+import '../../../../../shared/theme.dart';
+import '../../../../../shared/utils.dart';
+import '../../../shared/heap/class_filter.dart';
+import '../controller/utils.dart';
+
+class ClassFilterDialog extends StatefulWidget {
+  const ClassFilterDialog(this.classFilter, {super.key});
+
+  final ValueNotifier<ClassFilter> classFilter;
+
+  @override
+  State<ClassFilterDialog> createState() => _ClassFilterDialogState();
+}
+
+class _ClassFilterDialogState extends State<ClassFilterDialog> {
+  bool _initialized = false;
+  bool _showHelp = false;
+  late String _rootPackage;
+
+  late ClassFilterType _type;
+  final _except = TextEditingController();
+  final _only = TextEditingController();
+
+  @override
+  void initState() {
+    super.initState();
+    unawaited(_initialize());
+  }
+
+  Future<void> _initialize() async {
+    assert(!_initialized);
+    _rootPackage = await tryToDetectRootPackage() ?? '';
+    _loadStateFromFilter(widget.classFilter.value);
+    setState(() => _initialized = true);
+  }
+
+  @override
+  void didUpdateWidget(covariant ClassFilterDialog oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (oldWidget.classFilter.value != widget.classFilter.value) {
+      _loadStateFromFilter(widget.classFilter.value);
+    }
+  }
+
+  void _loadStateFromFilter(ClassFilter filter) {
+    _type = filter.filterType;
+    _except.text = filter.except;
+    _only.text = filter.only ?? _rootPackage;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+    final textFieldLeftPadding = scaleByFontFactor(40.0);
+    final itemSpacing = scaleByFontFactor(28.0);
```

Comment:
can we use shared values from theme.dart instead of defining our own here? For example we should use `defaultSpacing` instead of 28.0 to maintain consistency

Reply:
used defaultSpacing

But, I cannot replace textFieldLeftPadding as it should be of width of the radio button.
I think it is too specific for this dialog to share.